lib/pages/notifications/notification_service.dart

```diff
@@ -0,0 +1,127 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_local_notifications/flutter_local_notifications.dart';
+import 'package:flutter_timezone/flutter_timezone.dart';
+import 'package:get_it/get_it.dart';
+import 'package:mazilon/util/logger_service.dart';
+import 'package:timezone/data/latest_all.dart' as tz_data;
+import 'package:timezone/timezone.dart' as tz;
+import 'package:workmanager/workmanager.dart';
+
+class NotificationsService {
+  static bool _isInitialized = false;
+  static final FlutterLocalNotificationsPlugin
+      _flutterLocalNotificationsPlugin = FlutterLocalNotificationsPlugin();
+
+  static const InitializationSettings _initializationSettings =
+      InitializationSettings(
+    android: AndroidInitializationSettings(
+        '@mipmap/ic_launcher'), // Replace with your actual icon name
+    iOS: DarwinInitializationSettings(),
+  );
+
+  static Future<void> init() async {
+    if (_isInitialized) {
+      // Already initialized, no need to initialize again
+      return;
+    }
+
+    tz_data.initializeTimeZones();
+    // Get the device's current time zone
+    try {
+      final String timeZoneName = await FlutterTimezone.getLocalTimezone();
+      //  String currentTimeZone = await FlutterNativeTimezone.getLocalTimezone();
+
+      // Set the local time zone
+      tz.setLocalLocation(tz.getLocation(timeZoneName));
+      await _flutterLocalNotificationsPlugin
+          .initialize(_initializationSettings);
+      _isInitialized = true;
+    } catch (error, stackTrace) {
+      tz.setLocalLocation(tz.getLocation('Asia/Jerusalem'));
+      await _flutterLocalNotificationsPlugin
+          .initialize(_initializationSettings);
+      _isInitialized = true;
+      IncidentLoggerService loggerService =
+          GetIt.instance<IncidentLoggerService>();
+      await loggerService.captureException(
+        error,
+        stackTrace: stackTrace,
+      );
+    }
+  }
+
+  static Future<void> showNotification(String title, String body) async {
+    print("trying to show notification");
```

Comment:
Create a button under the current `save notification` button that will show an example of notification